### PR TITLE
♻️ refactor RUT to use new transactional context

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ♻️ refactor RUT to use new transactional context
  - Removed `ResourceTrackerRepository` and rather introduced three db modules: `credit_transactions_db`, `pricing_plans_db`, `service_runs_db` 

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                last_heartbeat_at=data.last_heartbeat_at,
+                modified=sa.func.now(),
+            )
+            .returning(resource_tracker_credit_transactions.c.transaction_id)
+        )
+        result = await conn.execute(insert_stmt)
+    row = result.first()
+    if row is None:
+        raise CreditTransactionNotCreatedDBError(data=data)
+    return cast(CreditTransactionId, row[0])
```
Comment (pcrespov) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:61]: TIP: you can use `row.transaction_id` and is more readable

**Review by pcrespov — APPROVED** [review]

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                service_run_id=data.service_run_id,
+                payment_transaction_id=data.payment_transaction_id,
+                created=data.created_at,
+                last_heartbeat_at=data.last_heartbeat_at,
+                modified=sa.func.now(),
+            )
+            .returning(resource_tracker_credit_transactions.c.transaction_id)
+        )
+        result = await conn.execute(insert_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:58]: I will point out that you are using the non blocking version of the cursor here. You need to await this.

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                )
+                & (
+                    resource_tracker_credit_transactions.c.last_heartbeat_at
+                    <= data.last_heartbeat_at
+                )
+            )
+            .returning(resource_tracker_credit_transactions.c.service_run_id)
+        )
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:95]: this is also blocking you asyncioloop

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                )
+                & (
+                    resource_tracker_credit_transactions.c.transaction_status
+                    == CreditTransactionStatus.PENDING
+                )
+            )
+            .returning(resource_tracker_credit_transactions.c.service_run_id)
+        )
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:128]: here as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                resource_tracker_credit_transactions.c.transaction_status.in_(
+                    [
+                        CreditTransactionStatus.BILLED,
+                        CreditTransactionStatus.PENDING,
+                    ]
+                )
+            )
+        )
+        result = await conn.execute(sum_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:157]: and here

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+            )
+            .order_by(
+                _version(
+                    resource_tracker_pricing_plan_to_service.c.service_version
+                ).desc()
+            )
+            .limit(1)
+        )
+        result = await conn.execute(query)
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:105]: and here

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 7]
```diff
@@ -0,0 +1,668 @@
...
+                & (resource_tracker_pricing_plans.c.product_name == product_name)
+                & (resource_tracker_pricing_plans.c.is_active.is_(True))
+            )
+            .order_by(resource_tracker_pricing_plan_to_service.c.pricing_plan_id.desc())
+        )
+        result = await conn.execute(query)
+
+    return [
+        PricingPlansWithServiceDefaultPlanDB.model_validate(row)
+        for row in result.fetchall()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:148]: this is also a blocking cursor for multiple entries

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 8]
```diff
@@ -0,0 +1,668 @@
...
+            resource_tracker_pricing_plans.c.classification,
+            resource_tracker_pricing_plans.c.is_active,
+            resource_tracker_pricing_plans.c.created,
+            resource_tracker_pricing_plans.c.pricing_plan_key,
+        ).where(
+            (resource_tracker_pricing_plans.c.pricing_plan_id == pricing_plan_id)
+            & (resource_tracker_pricing_plans.c.product_name == product_name)
+        )
+        result = await conn.execute(select_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:173]: here as well blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 9]
```diff
@@ -0,0 +1,668 @@
...
+            resource_tracker_pricing_plans.c.display_name,
+            resource_tracker_pricing_plans.c.description,
+            resource_tracker_pricing_plans.c.classification,
+            resource_tracker_pricing_plans.c.is_active,
+            resource_tracker_pricing_plans.c.created,
+            resource_tracker_pricing_plans.c.pricing_plan_key,
+        ).where(resource_tracker_pricing_plans.c.product_name == product_name)
+        result = await conn.execute(select_stmt)
+
+    return [PricingPlansDB.model_validate(row) for row in result.fetchall()]
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:197]: this is also blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 10]
```diff
@@ -0,0 +1,668 @@
...
+                    resource_tracker_pricing_plans.c.description,
+                    resource_tracker_pricing_plans.c.classification,
+                    resource_tracker_pricing_plans.c.is_active,
+                    resource_tracker_pricing_plans.c.created,
+                    resource_tracker_pricing_plans.c.pricing_plan_key,
+                ]
+            )
+        )
+        result = await conn.execute(insert_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:232]: here as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 11]
```diff
@@ -0,0 +1,668 @@
...
+                    resource_tracker_pricing_plans.c.description,
+                    resource_tracker_pricing_plans.c.classification,
+                    resource_tracker_pricing_plans.c.is_active,
+                    resource_tracker_pricing_plans.c.created,
+                    resource_tracker_pricing_plans.c.pricing_plan_key,
+                ]
+            )
+        )
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:274]: and here

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 12]
```diff
@@ -0,0 +1,668 @@
...
+            )
+            .where(
+                (resource_tracker_pricing_plans.c.product_name == product_name)
+                & (resource_tracker_pricing_plans.c.pricing_plan_id == pricing_plan_id)
+            )
+            .order_by(resource_tracker_pricing_plan_to_service.c.pricing_plan_id.desc())
+        )
+        result = await conn.execute(query)
+
+        return [PricingPlanToServiceDB.model_validate(row) for row in result.fetchall()]
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:317]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 13]
```diff
@@ -0,0 +1,668 @@
...
+                    == service_key
+                )
+                & (
+                    resource_tracker_pricing_plan_to_service.c.service_version
+                    == service_version
+                )
+            )
+        )
+        result = await conn.execute(query)
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:360]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 14]
```diff
@@ -0,0 +1,668 @@
...
+                *[
+                    resource_tracker_pricing_plan_to_service.c.pricing_plan_id,
+                    resource_tracker_pricing_plan_to_service.c.service_key,
+                    resource_tracker_pricing_plan_to_service.c.service_version,
+                    resource_tracker_pricing_plan_to_service.c.created,
+                ]
+            )
+        )
+        result = await conn.execute(insert_stmt)
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:396]: blocks as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 15]
```diff
@@ -0,0 +1,668 @@
...
+            )
+            .where(
+                (resource_tracker_pricing_units.c.pricing_plan_id == pricing_plan_id)
+                & (resource_tracker_pricing_unit_costs.c.valid_to.is_(None))
+            )
+            .order_by(resource_tracker_pricing_unit_costs.c.cost_per_unit.asc())
+        )
+        result = await conn.execute(query)
+
+    return [PricingUnitsDB.model_validate(row) for row in result.fetchall()]
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:460]: this one blocks as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 16]
```diff
@@ -0,0 +1,668 @@
...
+            .where(
+                (resource_tracker_pricing_units.c.pricing_plan_id == pricing_plan_id)
+                & (resource_tracker_pricing_units.c.pricing_unit_id == pricing_unit_id)
+                & (resource_tracker_pricing_unit_costs.c.valid_to.is_(None))
+                & (resource_tracker_pricing_plans.c.product_name == product_name)
+            )
+        )
+        result = await conn.execute(query)
+
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:504]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 17]
```diff
@@ -0,0 +1,668 @@
...
+                unit_extra_info=data.unit_extra_info.model_dump(),
+                default=data.default,
+                specific_info=data.specific_info.model_dump(),
+                created=sa.func.now(),
+                modified=sa.func.now(),
+            )
+            .returning(resource_tracker_pricing_units.c.pricing_unit_id)
+        )
+        result = await conn.execute(insert_stmt)
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:537]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 18]
```diff
@@ -0,0 +1,668 @@
...
+                valid_from=sa.func.now(),
+                valid_to=None,
+                created=sa.func.now(),
+                comment=data.comment,
+                modified=sa.func.now(),
+            )
+            .returning(resource_tracker_pricing_unit_costs.c.pricing_unit_cost_id)
+        )
+        result = await conn.execute(insert_stmt)
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:560]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 19]
```diff
@@ -0,0 +1,668 @@
...
+                    valid_from=sa.func.now(),
+                    valid_to=None,  # <-- New price is valid
+                    created=sa.func.now(),
+                    comment=data.pricing_unit_cost_update.comment,
+                    modified=sa.func.now(),
+                )
+                .returning(resource_tracker_pricing_unit_costs.c.pricing_unit_cost_id)
+            )
+            result = await conn.execute(insert_stmt)
+            row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:628]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py [hunk 20]
```diff
@@ -0,0 +1,668 @@
...
+            resource_tracker_pricing_unit_costs.c.created,
+            resource_tracker_pricing_unit_costs.c.comment,
+            resource_tracker_pricing_unit_costs.c.modified,
+        ).where(
+            resource_tracker_pricing_unit_costs.c.pricing_unit_cost_id
+            == pricing_unit_cost_id
+        )
+        result = await conn.execute(query)
+
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/pricing_plans_db.py:663]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 21]
```diff
@@ -0,0 +1,622 @@
...
+                started_at=data.started_at,
+                stopped_at=None,
+                service_run_status=ServiceRunStatus.RUNNING,
+                modified=sa.func.now(),
+                last_heartbeat_at=data.last_heartbeat_at,
+            )
+            .returning(resource_tracker_service_runs.c.service_run_id)
+        )
+        result = await conn.execute(insert_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:88]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 22]
```diff
@@ -0,0 +1,622 @@
...
+                )
+                & (
+                    resource_tracker_service_runs.c.last_heartbeat_at
+                    <= data.last_heartbeat_at
+                )
+            )
+            .returning(sa.literal_column("*"))
+        )
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:122]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 23]
```diff
@@ -0,0 +1,622 @@
...
+                (resource_tracker_service_runs.c.service_run_id == data.service_run_id)
+                & (
+                    resource_tracker_service_runs.c.service_run_status
+                    == ServiceRunStatus.RUNNING
+                )
+            )
+            .returning(sa.literal_column("*"))
+        )
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:153]: blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 24]
```diff
@@ -0,0 +1,622 @@
...
+    connection: AsyncConnection | None = None,
+    *,
+    service_run_id: ServiceRunId,
+) -> ServiceRunDB | None:
+    async with transaction_context(engine, connection) as conn:
+        stmt = sa.select(resource_tracker_service_runs).where(
+            resource_tracker_service_runs.c.service_run_id == service_run_id
+        )
+        result = await conn.execute(stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:170]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 25]
```diff
@@ -0,0 +1,622 @@
...
+                query = query.order_by(sa.asc(order_by.field))
+            else:
+                query = query.order_by(sa.desc(order_by.field))
+        else:
+            # Default ordering
+            query = query.order_by(resource_tracker_service_runs.c.started_at.desc())
+
+        result = await conn.execute(query)
+
+    return [ServiceRunWithCreditsDB.model_validate(row) for row in result.fetchall()]
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:294]: is is also blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 26]
```diff
@@ -0,0 +1,622 @@
...
+        # Default ordering and pagination
+        list_query = (
+            base_query.order_by(resource_tracker_service_runs.c.service_key.asc())
+            .offset(offset)
+            .limit(limit)
+        )
+        list_result = await conn.execute(list_query)
+
+    return (
+        cast(int, count_result.scalar()),
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:390]: this part is blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 27]
```diff
@@ -0,0 +1,622 @@
...
+        [
+            OsparcCreditsAggregatedByServiceKeyDB.model_validate(row)
+            for row in list_result.fetchall()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:393]: and of course here. both queries are blocking

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 28]
```diff
@@ -0,0 +1,622 @@
...
+
+        result = await conn.execute(
+            sa.DDL(
+                f"""
+            SELECT * from aws_s3.query_export_to_s3('{compiled_query}',
+            aws_commons.create_s3_uri('{s3_bucket_name}', '{s3_key}', '{s3_region}'), 'format csv, HEADER true');
+            """  # noqa: S608
+            )
+        )
+        row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:489]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 29]
```diff
@@ -0,0 +1,622 @@
...
+                sa.func.DATE(resource_tracker_service_runs.c.started_at)
+                <= started_until.date()
+            )
+        if service_run_status:
+            query = query.where(
+                resource_tracker_service_runs.c.service_run_status == service_run_status
+            )
+
+        result = await conn.execute(query)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:537]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 30]
```diff
@@ -0,0 +1,622 @@
...
+                resource_tracker_service_runs.c.service_run_status
+                == ServiceRunStatus.RUNNING
+            )
+            .order_by(resource_tracker_service_runs.c.started_at.desc())  # NOTE:
+            .offset(offset)
+            .limit(limit)
+        )
+        result = await conn.execute(query)
+
+    return [ServiceRunForCheckDB.model_validate(row) for row in result.fetchall()]
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:569]: here it is blocking as well

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 31]
```diff
@@ -0,0 +1,622 @@
...
+        query = (
+            sa.select(sa.func.count())
+            .select_from(resource_tracker_service_runs)
+            .where(
+                resource_tracker_service_runs.c.service_run_status
+                == ServiceRunStatus.RUNNING
+            )
+        )
+        result = await conn.execute(query)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:585]: blocks

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py [hunk 32]
```diff
@@ -0,0 +1,622 @@
...
+                & (
+                    resource_tracker_service_runs.c.last_heartbeat_at
+                    == last_heartbeat_at
+                )
+            )
+            .returning(sa.literal_column("*"))
+        )
+
+        result = await conn.execute(update_stmt)
+    row = result.first()
```
Comment (GitHK) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/service_runs_db.py:619]: blocks

**Review by GitHK — COMMENTED** [review]
Please be very carful when doing these changes. You have correctly awaited the execute part of the query, which is correct. But after an execute you are returned a cursor.

Each time you interact with a cursor it communicates to the DB. 
The operations `scalar`, `first`, `fetch_all` they all interact with the cursor. Please replace them with he `await` versions, which usually just implies in awaiting on the same method.

**Comment by matusdrobuliak66** [comment]
> Please be very carful when doing these changes. You have correctly awaited the execute part of the query, which is correct. But after an execute you are returned a cursor.
> 
> Each time you interact with a cursor it communicates to the DB. The operations `scalar`, `first`, `fetch_all` they all interact with the cursor. Please replace them with he `await` versions, which usually just implies in awaiting on the same method.

Discussed in person. There are two approaches: either the one I have, which buffers the results into memory, or the one that streams the results, where fetching also needs to be awaited.

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py [hunk 33]
```diff
@@ -0,0 +1,162 @@
...
+                user_id=data.user_id,
+                user_email=data.user_email,
+                osparc_credits=data.osparc_credits,
+                transaction_status=data.transaction_status,
+                transaction_classification=data.transaction_classification,
+                service_run_id=data.service_run_id,
+                payment_transaction_id=data.payment_transaction_id,
+                created=data.created_at,
+                last_heartbeat_at=data.last_heartbeat_at,
+                modified=sa.func.now(),
```
Comment (sanderegg) [inline services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:53]: if modified is set via the usual scripts, this is not necessary and same for created

**Review by sanderegg — APPROVED** [review]

File: services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py (hunk 33 above)
Comment (matusdrobuliak66) [reply services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/credit_transactions_db.py:53]: Yes, I prefer to do it manually in the DB layer whenever I need to (there are situations where I do not want to update it)

**Pull request merged**